lib/scholar/covariance.ex

```diff
@@ -27,6 +27,7 @@ defmodule Scholar.Covariance do
   @opts_schema NimbleOptions.new!(opts)
 
   @doc """
+  This function is deprecated and will be removed in the future. Use `Nx.covariance/2` instead.
```

Comment:
The correct way of deprecating is:

```
@deprecated "Use Nx.convariance/2 instead"
```

The compiler will also warn when people use the other ones.